pkg/scheduler/eventhandlers.go

```diff
@@ -176,11 +176,15 @@ func (sched *Scheduler) addPodToSchedulingQueue(obj interface{}) {
 }
 
 func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
-	pod := newObj.(*v1.Pod)
-	if sched.skipPodUpdate(pod) {
+	newPod := newObj.(*v1.Pod)
+	if sched.skipPodUpdate(newPod) {
 		return
 	}
-	if err := sched.SchedulingQueue.Update(oldObj.(*v1.Pod), pod); err != nil {
+	oldPod := oldObj.(*v1.Pod)
+	if oldPod.ResourceVersion == newPod.ResourceVersion {
+		return
+	}
```

Comment:
Should we test the version before invoking `sched.skipPodUpdate`, then we save some useless process.

Reply:
I thought of that as well. My idea was to make it conservative. But you're right, moving it ahead will be more efficient.

The key question is: can we get a consensus to compare ResourceVersion and do an early return?

@alculquicondor @ahg-g @damemi @adtac ^^

Reply:
I don't think doing this check after `skipPodUpdate` is any more conservative. So, fine doing it earlier.

Does ResourceVersion also change if there is a Status update?

Reply:
skipPodUpdate zeroes out ResourceVersion when comparing with in-cache copy. This seems to suggest that resource version may change even if there isn't a real pod update, right?

Reply:
That's not the intention of that zeroing. It's just that resource version will change when fields we don't care about change.

Reply:
> Does ResourceVersion also change if there is a Status update?

Yes, even with status update.

Reply:
> This seems to suggest that resource version may change even if there isn't a real pod update, right?

Nope, once any field of an obj is updated, its resource version gets changed.

One case that resource doesn't change is  "resync", the semantics of "resync" is to re-play the items in informerCache every "resync period", so it happens on client side, not server side. Those events individually carry identical objects actually. In some cases, some of them become a "phantom" event, just like this issue described.
I _hope_ that "resync" is the only case, so that #82222 should be good enough to handle this case. But I'm not 100% sure "resync" is the only culprit, so this PR is for extra guarding.